dart/colfer_gen_test.dart

```diff
@@ -0,0 +1,160 @@
+        O(s_: ' ' * 128)),
+    Golden('0901ff7f', O(a_: Uint8List.fromList([MaxUint8]))),
+    Golden('090202007f', O(a_: Uint8List.fromList([2, 0]))),
+    Golden(
+        '09c0010909090909090909090909090909090909090909090909090909090909090909090909090909090909090909090909090909090909090909090909090909090909090909090909090909090909090909090909090909090909090909090909090909090909090909090909090909090909090909090909090909090909090909090909090909090909090909090909090909090909090909090909090909090909090909090909090909090909090909090909090909090909090909090909097f',
+        O(a_: tmp)),
+    Golden('0a7f7f', O(o: O())),
+    Golden('0a007f7f', O(o: O(b: true))),
+    Golden('0b01007f7f', O(os: [O(b: true)])),
+    Golden('0b027f7f7f', O(os: [O(), O()])),
+    Golden('0c0300016101627f', O(ss: ['', 'a', 'b'])),
+    Golden(
+        '0d0201000201027f',
+        O(as_: [
+          Uint8List.fromList([0]),
+          Uint8List.fromList([1, 2])
+        ])),
+    Golden('0e017f', O(u8: 1)),
+    Golden('0eff7f', O(u8: MaxUint8)),
+    Golden('8f017f', O(u16: 1)),
+    Golden('0fffff7f', O(u16: MaxUint16)),
+    Golden('1002000000003f8000007f', O(f32s: Float32List.fromList([0, 1]))),
+    Golden('11014058c000000000007f', O(f64s: Float64List.fromList([99]))),
+  ];
+}
+
+void main() {
+  var cases = newGoldenCases();
+  group('Marshal', () {
+    for (final gold in cases) {
+      var buf = BytesBuilder(copy: false);
+      gold.object.marshal(buf);
+      test('marsaled binary should match golden string', () {
+        List<int> list = hex.decode(gold.serial);
+        Uint8List bytes = Uint8List.fromList(list);
+        expect(buf.toBytes(), bytes);
+      });
+    }
+  });
+
+  group('Unmarshal', () {
+    for (final gold in cases) {
+      test('struct should match golden string', () {
+        List<int> list = hex.decode(gold.serial);
+        Uint8List bytes = Uint8List.fromList(list);
+        var obj = O();
+        obj.unmarshal(bytes);
+        // work around NaN != NaN
+        if (obj.f32.isNaN && gold.object.f32.isNaN) {
+          obj.f32 = 0;
+          gold.object.f32 = 0;
+        }
+        if (obj.f64.isNaN && gold.object.f64.isNaN) {
+          obj.f64 = 0;
+          gold.object.f64 = 0;
+        }
+        expect(obj, equals(gold.object));
+      });
+    }
+  });
+
+  group('Unmarshal RangeError', () {
+    for (final gold in cases) {
+      test('should fail with RangeError', () {
+        List<int> list = hex.decode(gold.serial);
+        var obj = O();
+        for (int i = 0; i < list.length; i++) {
+          Uint8List bytes = Uint8List.fromList(list.sublist(0, i));
+          try {
+            obj.unmarshal(bytes);
+          } on RangeError {
+            continue;
+          } catch (e) {
+            print('something else than expected RangeError');
```

Comment:
The new `fail` still looses the e(rror) information. In general—not just for testing—you can only ignore errors if you have identified which one it is, and you know how to act upon it. In this case we can just print e in the fail message. Do you agree?

Reply:
`print(e)` gives a syntax error (need string), so I ended up with a short `got "$e" instead of expected RangeError` description. Feel free to change if too verbose :)

Reply:
I expect `fail("want a RangeError, got: " + e.toString())` to do the job.

Reply:
Ok. Dart has this string interpolation, so printing `want a RangeError, got: $e` does the same thing.
https://shailen.github.io/blog/2012/11/14/dart-string-interpolation/